PizzaBuilder/PizzaBuilder/Program.cs

```diff
@@ -0,0 +1,188 @@
+            return result;
+        }
+    }
+
+    public class WhitePizza
+    {
+        private List<object> pizza = new List<object>();
+        public void Add(string part)
+        {
+            this.pizza.Add(part);
+        }
+        public string ReadyPizza()
+        {
+            string str = "";
+            for (int i = 0; i < this.pizza.Count; i++)
+            {
+                str += this.pizza[i];
+            }
+            return "White Pizza ingredients : " + str;
+        }
+
+    }
+
+    public class WheatPizza
+    {
+        private List<object> pizza = new List<object>();
+        public void Add(string part)
+        {
+            this.pizza.Add(part);
+        }
+        public string ReadyPizza()
+        {
+            string str = "";
+            for (int i = 0; i < this.pizza.Count; i++)
+            {
+                str += this.pizza[i];
+            }
+            return "Wheat Pizza ingredients : " + str;
+        }
+
+    }
+
+
+    public class Director
+    {
+        private IBuilder _whitepizza;
+        public IBuilder whitepizzaBuilder
+        {
+            set { this._whitepizza = value; }
+        }
+
+        public void BuildMin_white()
+        {
+            this._whitepizza.BuildDough();
+            this._whitepizza.BuildSauce();
+        }
+        public void BuildFull_white()
+        {
+            this._whitepizza.BuildDough();
+            this._whitepizza.BuildSauce();
+            this._whitepizza.BuildTopping();
+        }
+
+        private IBuilder _wheatpizza;
+        public IBuilder wheatpizzaBuilder
+        {
+            set { this._wheatpizza = value; }
+        }
+        public void BuildMin_wheat()
+        {
+            this._wheatpizza.BuildDough();
+            this._wheatpizza.BuildSauce();
+        }
+        public void BuildFull_wheat()
+        {
+            _wheatpizza.BuildDough();
+            this._wheatpizza.BuildSauce();
+            this._wheatpizza.BuildTopping();
+        }
+    }
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("-*-*- white Yummy pizza -*-*-");
+            //client
+            var director_1 = new Director();
+            var whitepizza_builder = new WhitePizzaBuilder();
+            director_1.whitepizzaBuilder = whitepizza_builder;
+            //Full Product
+            director_1.BuildFull_white();
+            Console.WriteLine(whitepizza_builder.GetWhitePizza().ReadyPizza());
+            //Min Product
+            director_1.BuildMin_white();
+            Console.WriteLine(whitepizza_builder.GetWhitePizza().ReadyPizza());
+            //Custom Product
+            whitepizza_builder.BuildDough();
+            whitepizza_builder.BuildTopping();
+            Console.WriteLine(whitepizza_builder.GetWhitePizza().ReadyPizza());
+
+            Console.WriteLine("-*-*- wheat Yummy pizza -*-*-");
+            //client
+            var director_2 = new Director();

```

Comment:
You don't need to declare the Director two times just declare it one time and use it with any Builder